File: JSON_ODF_reader_plotsample _MTR.py
'''
Created on 2017-10-26 by Dave Senciall
@author: DSenciall

NOTES:
coded for python 3.x : the panda's to_xx functions dont seem to be present in the anancoda 2.7 for some
bizzare reason even though they are part of pandas since pandas 0.17

'''

import datetime
#import re
from collections import OrderedDict
import json
import pandas as pd
import matplotlib.pyplot as pyplot


def read_JSON(afile):
    jdict = OrderedDict()
    try:
        with open(afile, 'r') as fp:
            jdict = json.load(fp)

        fp.close()
    except:
        pass
    return (jdict)

def print_text(adict):
        for i in adict :
            for j in adict[i]:
                if not isinstance(j, list):
                    print (i,j, adict[i][j])
                else:
#                    for k  in j:
                        print (j)


def main():

# reasd JSON file that has encode ODF

    hdr = read_JSON("MTR_HUD2015030_1898_10588081_1800_ODF.json")


# demo of accessing a header element value
    print (hdr["RECORD_HEADER"],hdr["RECORD_HEADER"]["NUM_CYCLE"])

# get the data block (list of lists)	
    data = hdr["DATA"]

# coerce it to a pandas data frame
    pd_data = pd.DataFrame(data)

# label the channels (technicall you could pull the channel ids fromt he parameters blocks)	
    pd_data.columns=['DateTime','Temp']

# set the data types for the columns
    pd_data['DateTime']= pd.to_datetime(pd_data['DateTime'],format='%d-%b-%Y %H:%M:%S.00')
    pd_data["Temp"] = pd.to_numeric(pd_data["Temp"])

#plot the data
    pd_data.plot(x='DateTime',y='Temp',title='BIO HOBO demo')
    pyplot.show()




if __name__ == "__main__":
    main()

File: JSON_ODF_reader_plotsample_MTR.r
#Created on 2017-11-01 by Dave Senciall
#@author: DSenciall

#Install package
#install.packages("rjson")  only needed if not already on yo8r machine

library(rjson)


jsonfile <- "C:\\sources\\GitHub\\Data_containers\\MTR_HUD2015030_1898_10588081_1800_ODF.json" 



 hdr<- fromJSON(file = jsonfile)

datalength <- length(hdr$DATA)

# see how many lines you got of data
datalength


#atributes (hdr)

# and a demo of accessing a header element value
hdr$RECORD_HEADER$NUM_CYCLE


# convert it to dataframe
# unlike in python for R need to transposed to get cols and rows right
hdrdata= as.data.frame(t(as.data.frame(hdr$DATA))) 

#hdrdata= as.data.frame(hdr$DATA) # transposed to get cols and rows right


# check the attributes
#str(hdrdata)

# label the channels (technicall you could pull the channel ids fromt he parameters blocks)	 
colnames(hdrdata) <- c('date','temp')


# conver to a date ;convert temp from an R 'factor' to a numeric
hdrdata['date'] <-  as.POSIXct(hdrdata[,'date'],format = "%d-%b-%Y %H:%M:%S.00", tz="")
hdrdata['temp'] <-  as.numeric(paste(hdrdata[,'temp']))


# plot the data
library(scales)
require(ggplot2)


# note : for the aes call do not put variable names in quotes
aplot<- ggplot(hdrdata, aes(x =date, y = temp)) +
        geom_line() +
        scale_y_continuous(name='temp',limits=c(-2,6))

aplot

                     
